proposals/provenance-logging.md

```diff
@@ -0,0 +1,128 @@
+## Motivation
+Reproducible ML is a [big problem](https://petewarden.com/2018/03/19/the-machine-learning-reproducibility-crisis/). 
+Kubeflow could add a lot of values to users by making it easy for users to track how datasets and models were produced.
+A lot of valuable information is available in K8s resource specifications. Resources like K8s Jobs, TFJobs, ConfigMaps, and Argo workflows
```

Comment:
"This would go beyond just data provenance and job specification; we should also capture the valuable information in setup, including resources like..."